### DISCUSSION TITLE
Layouts extraction options in iterdxf

### DISCUSSION THREAD

**Discussion by IvanKachaikinCendas** [mozman/ezdxf#823] [Q&A] [answered]
Hi @mozman!

First of all, just once again wanted to say thanks for your amazing package!

This one is not really an issue, rather a question. In `iterdxf` add-on it is possible to get an access to the model space layout according to the [documentation](https://ezdxf.readthedocs.io/en/stable/addons/iterdxf.html#module-ezdxf.addons.iterdxf). Is there a way to get an access to other layouts (paperspaces) in the drawing?

Thanks in advance for your feedback!

**Comment by mozman** [comment, marked as answer]
The current implementation supports only the modelspace. The `iterdxf.IterDXF` class and the function `iterdxf.opendxf()` which is based on `IterDXF` are hard to extend to load paperspace layouts, because they always create a fully valid DXF file which also includes the modelspace and the management of paperspace layouts requires parsing of structures distributed across the TABLES, ENTITIES and OBJECTS sections.

It's possible to add a function `iterdxf.paperspace()` which works like the `iterdxf.modelspace()`  function and yields the content entities of the **active** paperspace which is determined by the BLOCK_RECORD named "*Paper_Space". It would even be possible to yield the content of any paperspace layout, but you have to know the BLOCK_RECORD name of the layout (e.g. "*Paper_Space0", "*Paper_Space1" and so on), because detecting the layout name as displayed in the CAD application requires parsing of structures distributed across the TABLES and OBJECTS sections.

I am not sure if the content of a paperspace layout is even useful, you don't get the content of VIEWPORT entities, you get **the** VIEWPORT entity which defines the region of the modelspace which should be rendered by the DXF viewer or CAD application.

**Reply by IvanKachaikinCendas** [reply]
Hi @mozman! Ok, that seems to be clear, thanks a lot for your feedback!
